airflow/models/dag.py

```diff
@@ -678,6 +678,12 @@ def __init__(
             self.timetable = DatasetTriggeredTimetable(DatasetAll(*schedule))
             self.schedule_interval = self.timetable.summary
         elif isinstance(schedule, ArgNotSet):
+            warnings.warn(
```

Comment:
Just curious - maybe lagging history: Why was it never considered setting `schedule=None` per default?
If I take a look to ALL your changes in all tests I see this seems the be actually a common and reasonable default (whereas I acknowledge thatchanging the default in a breaking change might confuse and irritate a LOT of users)

Reply:
We've definitely considered it and wanted to do it for a long time now. But yeah, breaking change...

Reply:
Yeah there is already a default, and we can’t really just change that default unless we warn everyone not use that default first. I do wonder if there’s a better way.

Reply:
Yeah... no better way... only chance would be to drop the attribute (which we did in the past once :-D) and add a new one with a new default... yeah. sad.